src/decision-engine/index.js

```diff
@@ -1,110 +1,126 @@
 'use strict'
 
-const debounce = require('just-debounce-it')
+const CID = require('cids')
 
 const Message = require('../types/message')
+const WantType = Message.WantType
 const Wantlist = require('../types/wantlist')
 const Ledger = require('./ledger')
-const { logger, groupBy, pullAllWith, uniqWith } = require('../utils')
+const RequestQueue = require('./req-queue')
+const TaskMerger = require('./task-merger')
+const { logger } = require('../utils')
 
-const MAX_MESSAGE_SIZE = 512 * 1024
+// The ideal size of the batched payload. We try to pop this much data off the
+// request queue, but it may be a little more or less depending on what's in
+// the queue.
+const TARGET_MESSAGE_SIZE = 16 * 1024
+
+// The maximum size of the block in bytes up to which we will replace a
+// want-have with a want-block
+const MAX_BLOCK_SIZE_REPLACE_HAS_WITH_BLOCK = 1024
```

Comment:
Sorry this comment is badly written and probably caused some confusion further down.

The client sends want-have (I want a HAVE) or want-block (I want a block). If the client sends want-have and the block is small enough, we just respond with the block (instead of responding with a HAVE). The constant is the maximum size of a block such that we will replace a HAVE with a block.

I'll update the comment and the name of the constant.